compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -224,12 +224,17 @@ impl<'ast, 'ra: 'ast, 'tcx> LateResolutionVisitor<'_, 'ast, 'ra, 'tcx> {
                 let suggestion = if self.current_trait_ref.is_none()
                     && let Some((fn_kind, _)) = self.diag_metadata.current_function
                     && let Some(FnCtxt::Assoc(_)) = fn_kind.ctxt()
-                    && let FnKind::Fn(_, _, _, ast::Fn { sig, .. }) = fn_kind
+                    && let FnKind::Fn(_, _, ast::Fn { sig, .. }) = fn_kind
                     && let Some(items) = self.diag_metadata.current_impl_items
                     && let Some(item) = items.iter().find(|i| {
-                        i.ident.name == item_str.name
+                        if let Some(ident) = i.kind.ident()
```

Comment:
theoretically this could be made more terse:

```rs
i.kind.ident().is_some_and(|ident| ident.name == item_str.name && !sig.span.contains(item_span))
```

but I let you decide if you find that better / necessary